src/Catty/ViewController/Upload/UploadViewController.swift

```diff
@@ -259,171 +203,32 @@ class UploadViewController: UIViewController {
-                                 with: token.data(using: .utf8),
-                                 forHTTPBody: &body)
-
-            //Username
-            setFormDataParameter(userNameTag,
-                                 with: (UserDefaults.standard.value(forKey: kcUsername) as! String).data(using: .utf8),
-                                 forHTTPBody: &body)
-
-            //Language
-            setFormDataParameter(deviceLanguageTag,
-                                 with: NSLocale.preferredLanguages[0].data(using: .utf8),
-                                 forHTTPBody: &body)
-
-            //zip file
-            setAttachmentParameter(uploadParameterTag,
-                                   with: zipFileData,
-                                   forHTTPBody: &body)
-
-            // close request form
-            if let anEncoding = "--\(httpBoundary)--\r\n".data(using: .utf8) {
-                body.append(anEncoding)
-            }
-            // set request body
-            request.httpBody = body
-
-            let postLength = String(format: "%lu", UInt(body.count))
-            request.addValue(postLength, forHTTPHeaderField: "Content-Length")
-
-            // debug output
-            let string1 = String(data: request.httpBody!,
-                                 encoding: String.Encoding.utf8) ?? "Data could not be printed"
-            debugPrint(string1)
-
-            // start request
-            let dataTask = session!.dataTask(with: request as URLRequest) { data, _, error in
-                self.enableUploadView()
-                if error != nil {
-                    debugPrint("Connection could not be established")
-                    Util.defaultAlertForNetworkError()
-                } else {
-                    var dictionary: [AnyHashable: Any]?
-                    if let aData = data {
-                        do {
-                            dictionary = try JSONSerialization.jsonObject(with: aData, options: []) as? [AnyHashable: Any]
-                        } catch {
-                            print("JSON response could not be serialized")
-                        }
-                    }
-                    var statusCode: String?
-                    if let aTag = dictionary?[self.statusCodeTag] {
-                        statusCode = "\(aTag)"
-                    }
-                    debugPrint("StatusCode is ", statusCode ?? "undefined")
-
-                    if statusCode == self.statusCodeOK {
-                        debugPrint("Upload successful")
-
-                        //Set unique Project-ID received from server
-                        if let aTag = dictionary?[self.projectIDTag], let project = self.project {
-                            let projectId = "\(aTag)"
-
-                            DispatchQueue.main.async(execute: {
-                                self.project?.rename(toProjectName: project.header.programName, andProjectId: projectId, andShowSaveNotification: true)
```

Comment:
The final rename to the `projectId` is missing in the new implementation (please see me other comments).